Titanic_Test.R

```diff
@@ -1,165 +1,165 @@
-# 
-# #Check the total 'NA' datas
-# sum(is.na(titanic_test.df))
-# 
-# #Check the 'NA' datas according to the variables
-# sapply(titanic_test.df, function (x) sum(is.na(x)))
-# 
-# #Adding Survived variable to the test data with None value
-# titanic_test.df <- data.frame('Survived' = rep("None", nrow(titanic_test.df)), titanic_test.df[,])
-# 
-# #Cabin is not significant, hence dropping the column
-# titanic_test.df$Cabin <- NULL
-# 
-# titles <- NULL
-# for (i in 1:nrow(titanic_test.df)) {
-#   titles <- c(titles, extractTitle(titanic_test.df[i,"Name"], titanic_test.df[i,"Parch"]))
-# }
-# titanic_test.df$Title <- as.factor(titles)
-# 
-# titanic_test.df$Survived <- as.factor(titanic_test.df$Survived)
-# titanic_test.df$Pclass <- as.factor(titanic_test.df$Pclass)
-# #titanic_test.df$SibSp <- as.factor(titanic_test.df$SibSp)
-# #titanic_test.df$Parch <- as.factor(titanic_test.df$Parch)
-# str(titanic_test.df)
-# titanic_test.df$Age[is.na(titanic_test.df$Age)] <- median(titanic_test.df$Age, na.rm=TRUE)
-# titanic_test.df$Fare[is.na(titanic_test.df$Fare)] <- median(titanic_test.df$Fare, na.rm=TRUE)
-# 
-# familySize <- titanic_test.df$SibSp+titanic_test.df$Parch+1
-# titanic_test.df$FamilySize <- as.integer(familySize)
-# 
-# #Predicting the test data from the training model
-# titanic_predict <- c('Pclass', 'Title','FamilySize','Fare')
-# titanic_test_predict.df <- predict(titanic.glm2, titanic_test.df[,titanic_predict], type="response")
-# head(titanic_test_predict.df)
-# head(titanic_test.df)
-# 
-# y_pred = ifelse(titanic_test_predict.df > 0.5, 1, 0)
-# head(y_rf_pred)
-# 
-# output <- data.frame(cbind('PassengerId' = titanic_test.df$PassengerId, 'Survived' =y_pred))
-# write.csv(output, "output6.csv",row.names = FALSE)
-# 
-# #Random Forest Implementation
-# library(randomForest)
-# set.seed(1234)
-# rf <- randomForest(Survived ~ Pclass+Title+FamilySize+Fare, titanic.df)
-# 
-# predict(rf, titanic_test.df[,titanic_predict])
-# rf_predict <- predict(rf, titanic_test.df[,titanic_predict])
-# head(rf_predict)
-# 
-# output <- data.frame(PassengerId = titanic_test.df$PassengerId, Survived =rf_predict)
-# write.csv(output, "output13.csv",row.names = FALSE)
\ No newline at end of file
+setwd("E:/Masters/Semester 2/Data Mining CS6405/Titanic/Titanic/data/external")
+
+#Reading Data from the CSV file
+titanic.df <- read.csv(file="train.csv",header=TRUE, na.strings = "")
+
+#Check Dimension of the dataset
+dim(titanic.df)
+
+#Check the structure of the dataset
+str(titanic.df)
+
+#Check the total 'NA' datas
+sum(is.na(titanic.df))
+
+#Check the 'NA' datas according to the variables
+sapply(titanic.df, function (x) sum(is.na(x)))
+
+#Making the data variables as factors
+titanic.df$Survived <- as.factor(titanic.df$Survived)
+titanic.df$Pclass <- as.factor(titanic.df$Pclass)
+#titanic.df$SibSp <- as.factor(titanic.df$SibSp)
+#titanic.df$Parch <- as.factor(titanic.df$Parch)
+
+#Extract the title from the name
+extractTitle <- function(name, nop) {
```

Comment:
Using Regex will be easy and fast!!
title <- regmatches(titanic.df$Name, regexpr("\\,\\s\\w+", titanic.df$Name))
title <- gsub("\\,\\s","",title)
length(title) #891